### PR TITLE
Compare Command

### PR DESCRIPTION
**Pull request by typat**
added the ability to compare associated users in the channel. for now, shows set stats that are included on the lifetime stats view.

### CODE REVIEW

File: src/commands/compare-command.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+                if (!value.bot) {
+                    ids.push(key);
+                }
+            }
+            if (!ids.length) {
+
+            }
+            const userDb = new UserDB();
+            let docs = await userDb.getUsers(ids);
+            let stats = await Promise.all(docs.map(doc => this.API.MWstats(doc.username, doc.platform)));
```
Comment (typat) [inline src/commands/compare-command.js:35]: tried my best to combine all these, any suggestions here?

File: src/commands/compare-command.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+                obj.value = `${obj.value.toFixed(2)}`
+            }
+            temp.push(obj);
+        }
+        temp.sort((a, b) => b.value - a.value)
+        return temp.map(function (i) {
+            return `${i.name} (${i.value})`
+        })
+            .join(', ');
+    }
```
Comment (typat) [inline src/commands/compare-command.js:78]: any advice on cleaning this method up?

File: src/commands/compare-command.js (hunk 1 above)
Comment (cracker4o) [reply src/commands/compare-command.js:35]: I think you can do:
```
let stats = docs.map(async doc => await this.API.MWstats(doc.username, doc.platform)));
```
docs is not a promise and only the MWstats returns a promise which we wait for in the lambda.

File: src/commands/compare-command.js (hunk 2 above)
Comment (cracker4o) [reply src/commands/compare-command.js:78]: Maybe this, but I got to try it.
```
return stats.map(stat => {
              let obj = {
                name: stat.username.split('#')[0],
                value: _.get(stat, prop, "N/A")
            }
            if (fixed) {
                obj.value = `${obj.value.toFixed(2)}`
            }

            return obj;
}).sort((a, b) => b.value - a.value)
.map(item => `${item.name} (${item.value})`).join(', ');

**Review by cracker4o — APPROVED** [review]
I like it. This is a great feature. I'll merge it.

**Pull request merged**